catalyst/contrib/nn/criterion/cluster.py

```diff
@@ -0,0 +1,164 @@
+import torch
+from torch import nn
+
+from catalyst.contrib.nn.criterion.functional import euclidean_distance
+
+
+class ClusterLoss(nn.Module):
+    """
+    Cluster loss
+    .. _Cluster Loss for Person Re-Identification
+    https://arxiv.org/pdf/1812.10325.pdf
```

Comment:
is it possible to add minimal example for this loss?
in long term it's much easier to understand them

something like
```
embeddings = toch.sample
targets = torch.sample
criterion = ClusterLoss()
loss = criterion(embeddings, targets)
```

Reply:
it would be also nice to see some kind of tests for this criterion like
https://github.com/catalyst-team/catalyst/blob/master/catalyst/utils/metrics/tests/test_iou.py
to ensure it correctness :)